File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.py
a = int(input())

arr = []
for _ in range(a):
  arr.append(input())

temp = arr[0]

for i in range(len(temp)):
  flag = True
  for j in range(a):
    if arr[j][i] != temp[i]:
      flag = False

  if flag:
    print(temp[i], end = '')
  else:
    print("?", end = '')

File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   int a, flag;
   scanf("%d", &a);

   char str[a][51];
   for(int i = 0; i < a; i++){
      scanf("%s", str[i]);
   }

   char temp[51];
   strcpy(temp, str[0]);
   
   for(int i = 0; i < strlen(str[0]); i++){
      flag = 1;
      for(int j = 1; j < a; j++){
         if(str[j][i] != temp[i]){
            flag = 0;
         }
      }
      if(flag){
         printf("%c", temp[i]);
      }
      else{
         printf("?");
      }
   }
   
   return 0;
}